pkg/genericapiserver/server/openapi/openapi.go

```diff
@@ -79,6 +81,15 @@ func (o *openAPI) init(webServices []*restful.WebService) error {
 			return r.Operation, nil, nil
 		}
 	}
+	if o.config.GetDefinitionName == nil {
+		o.config.GetDefinitionName = func(_, name string) (string, []string) {
+			return name[strings.LastIndex(name, "/")+1:], nil
+		}
+	}
+	o.definitions = o.config.GetDefinitions(func(name string) spec.Ref {
+		defName, _ := o.config.GetDefinitionName(o.servePath, name)
+		return spec.MustCreateRef("#/definitions/" + openapi.EscapeJsonPointer(defName))
```

Comment:
We should verify that this doesn't horribly break any of the generation code we use today (docs, client code) - I am skeptical that people implementing those libraries handle escaping correctly, because no one handles escaping correctly.